app/endpoints/declared_endpoints.py

```diff
@@ -252,6 +258,134 @@ async def test_neuron_file(
+                        "OU10Me01", "SResetITC", "STrueNoise", "SubWhiteNoise",
+                        "Truenoise", "WhiteNoise", "ResetITC", "SponHold25", "SponHold3",
+                        "SponHold30", "SSponHold", "SponNoHold20", "SponNoHold30",
+                        "SSponNoHold", "Spontaneous", "hold_dep", "hold_hyp",
+                        "StartHold", "StartNoHold", "StartStandeCode", "VacuumPulses",
+                        "sAHP", "IRdepol", "IRhyperpol", "IDdepol", "IDhyperpol",
+                        "SsAHP", "HyperDePol", "DeHyperPol", "NegCheops", "NegCheops1",
+                        "NegCheops2", "NegCheops3", "NegCheops4", "NegCheops5",
+                        "PosCheops", "Rin_dep", "Rin_hyp", "SineSpec", "SSineSpec",
+                        "Pulse", "S2", "s2", "S30", "SIne20Hz", "A___.ibw"]
+
+    for readerclass in NWB_READERS:
+        try:
+            # 1. Initialize the reader with both file path AND target_protocols
+            reader = readerclass(nwb_file_path, test_protocols)
+
+            # 2. Attempt to read the data
+            data = reader.read()
+
+            # 3. Check for success (data is not None)
+            if data is not None:
+                return data
+
+        except Exception as e:  # noqa: BLE001
+            L.warning(
+                "Reader %s failed for file %s: %s",
+                readerclass.__name__,
+                nwb_file_path,
+                str(e)
+            )
+            continue
+
+    # If the loop finishes without returning, no reader worked.
+    # This is the point where we raise an error as requested.
+    reader_names = ", ".join([r.__name__ for r in NWB_READERS])
+    error_message = (
+        f"All {len(NWB_READERS)} NWB readers failed to read the file '{nwb_file_path}'.\n"
+        f"Attempted readers: {reader_names}"
+    )
+    raise RuntimeError(error_message)
+
+
+def activate_validate_nwb_endpoint(router: APIRouter) -> None:
+    """Define neuron file test endpoint."""
+
+    @router.post(
+        "/validate-nwb-file",
+        summary="Validate new format.",
+        description="Tests a new file (.nwb) with basic validation.",
+    )
+    async def validate_nwb_file(
+        file: Annotated[UploadFile, File(description="Nwb file to upload (.swc, .h5, or .asc)")],
+    ) -> FileResponse:
+        file_extension = f".{file.filename.split('.')[-1].lower()}" if file.filename else ""
+        temp_file_path = ""
+
+        # 1. Read content async (usually non-blocking)
+        content = await file.read()
+        if not content:
+            _handle_empty_file(file)
+
+        # Define a synchronous function to handle all blocking I/O (write + read)
+        def blocking_file_operations(file_content: bytes, suffix: str) -> str:
+            """Synchronously writes the file and runs the NWB reader."""
+            with tempfile.NamedTemporaryFile(delete=False, suffix=suffix) as temp_file:
```

Comment:
Is the suffix needed to make the file readable by the validators? In that case, it would be better to add a comment.

Reply:
The comment has been added.